pkg/controller/mongodb/mongodb_controller.go

```diff
@@ -265,24 +265,34 @@ func (r ReplicaSetReconciler) ensureAutomationConfig(mdb mdbv1.MongoDB) error {
 	if err != nil {
 		return err
 	}
+
 	if err := r.client.CreateOrUpdate(&cm); err != nil {
 		return err
 	}
 	return nil
 }
 
-func buildAutomationConfig(mdb mdbv1.MongoDB, mdbVersionConfig automationconfig.MongoDbVersionConfig) automationconfig.AutomationConfig {
+func buildAutomationConfig(mdb mdbv1.MongoDB, mdbVersionConfig automationconfig.MongoDbVersionConfig, currentAc automationconfig.AutomationConfig) automationconfig.AutomationConfig {
 	domain := getDomain(mdb.ServiceName(), mdb.Namespace, "")
-	return automationconfig.NewBuilder().
+	newAc := automationconfig.NewBuilder().
 		SetTopology(automationconfig.ReplicaSetTopology).
 		SetName(mdb.Name).
 		SetDomain(domain).
 		SetMembers(mdb.Spec.Members).
 		SetMongoDBVersion(mdb.Spec.Version).
-		SetAutomationConfigVersion(1). // TODO: Correctly set the version
+		SetAutomationConfigVersion(currentAc.Version).
 		SetFCV(mdb.GetFCV()).
 		AddVersion(mdbVersionConfig).
 		Build()
+
+	b1, _ := json.Marshal(newAc)
+	b2, _ := json.Marshal(currentAc)
```

Comment:
we should never ignore the errors! It looks like we'll need to update the method signature here to also return an error.

We can also name them a little better. Even `newAcBytes` and `existingAcBytes`